src/sketch.js

```diff
@@ -144,124 +255,77 @@ function draw() {
+    sesh.currentY -= speed;
   }
   if (keyIsDown(83) === true) {
     // S key; go down
-    activeCreature.y += speed;
-    storeItem(activeCreatureName + "Y", activeCreature.y);
+    sesh.currentY += speed;
   }
 
   // move the active creature by tapping a touchscreen
-  if (touchTarget) {
-    if (activeCreature.x > touchTarget.x) {
-      activeCreature.x -= 5;
-      storeItem(activeCreatureName + "X", activeCreature.x);
+  if (sesh.touchTarget) {
+    if (sesh.currentX > sesh.touchTarget.x) {
+      sesh.currentX -= 5;
     }
-    if (activeCreature.x < touchTarget.x) {
-      activeCreature.x += 5;
-      storeItem(activeCreatureName + "X", activeCreature.x);
+    if (sesh.currentX < sesh.touchTarget.x) {
+      sesh.currentX += 5;
     }
-    if (activeCreature.y > touchTarget.y) {
-      activeCreature.y -= 5;
-      storeItem(activeCreatureName + "Y", activeCreature.y);
+    if (sesh.currentY > sesh.touchTarget.y) {
+      sesh.currentY -= 5;
     }
-    if (activeCreature.y < touchTarget.y) {
-      activeCreature.y += 5;
-      storeItem(activeCreatureName + "Y", activeCreature.y);
+    if (sesh.currentY < sesh.touchTarget.y) {
+      sesh.currentY += 5;
     }
   }
 
   // keep creatures within the bounds of the screen
-  if (activeCreature.x < 0 - CREATURE_SIZE) {
-    activeCreature.x = windowWidth + CREATURE_SIZE;
-    storeItem(activeCreatureName + "X", activeCreature.x);
+  if (sesh.currentX < 0 - sesh.creatureSize) {
+    sesh.currentX = sesh.width + sesh.creatureSize;
   }
-  if (activeCreature.x > windowWidth + CREATURE_SIZE) {
-    activeCreature.x = 0 - CREATURE_SIZE;
-    storeItem(activeCreatureName + "X", activeCreature.x);
+  if (sesh.currentX > sesh.width + sesh.creatureSize) {
+    sesh.currentX = 0 - sesh.creatureSize;
   }
-  if (activeCreature.y < 0 - CREATURE_SIZE) {
-    activeCreature.y = windowHeight + CREATURE_SIZE;
-    storeItem(activeCreatureName + "Y", activeCreature.y);
+  if (sesh.currentY < 0 - sesh.creatureSize) {
+    sesh.currentY = sesh.height + sesh.creatureSize;
   }
-  if (activeCreature.y > windowHeight + CREATURE_SIZE) {
-    activeCreature.y = 0 - CREATURE_SIZE;
-    storeItem(activeCreatureName + "Y", activeCreature.y);
+  if (sesh.currentY > sesh.height + sesh.creatureSize) {
+    sesh.currentY = 0 - sesh.creatureSize;
   }
 }
 
 // switch between creatures using arrow keys
 function keyReleased() {
-  let i = getItem("activeCreatureIndex");
-
-  // toggle forwards
   if (key === "ArrowRight" || key === "ArrowUp") {
-    i = (i + 1) % CREATURES.size;
-    setActiveCreature(i);
+    sesh.nextActiveCreature();
   }
-
-  // toggle backwards
   if (key === "ArrowLeft" || key === "ArrowDown") {
-    if (i === 0) {
-      i = CREATURES.size - 1;
-    } else {
-      i = (i - 1) % CREATURES.size;
-    }
-    setActiveCreature(i);
+    sesh.lastActiveCreature();
   }
 }
 
-// reset creature position when window size changes
+// reset creature session when window size changes
+let timeout = false;
```

Comment:
should this be `timeoutId` or something? and undefined instead of a boolean

Reply:
tru i agree... changing it now